addons/binding/org.openhab.binding.onebusaway/src/main/java/org/openhab/binding/onebusaway/handler/RouteHandler.java

```diff
@@ -0,0 +1,278 @@
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.onebusaway.handler;
+
+import static org.openhab.binding.onebusaway.OneBusAwayBindingConstants.*;
+
+import java.util.Calendar;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.core.library.types.DateTimeType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.type.ChannelKind;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.onebusaway.config.ChannelConfig;
+import org.openhab.binding.onebusaway.config.RouteConfiguration;
+import org.openhab.binding.onebusaway.handler.ObaStopArrivalResponse.ArrivalAndDeparture;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Function;
+
+/**
+ * The {@link RouteHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Shawn Wilsher - Initial contribution
+ */
+public class RouteHandler extends BaseThingHandler implements RouteDataListener {
+
+    public static final ThingTypeUID SUPPORTED_THING_TYPE = THING_TYPE_ROUTE;
+
+    private RouteConfiguration config;
+    private Logger logger = LoggerFactory.getLogger(RouteHandler.class);
+    private List<ScheduledFuture<?>> scheduledFutures = new CopyOnWriteArrayList<>();
+
+    public RouteHandler(Thing thing) {
+        super(thing);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (RefreshType.REFRESH == command) {
+            logger.debug("Refreshing {}...", channelUID);
+            switch (channelUID.getId()) {
+                case CHANNEL_ID_ARRIVAL:
+                case CHANNEL_ID_DEPARTURE:
+                case CHANNEL_ID_UPDATE:
+                    StopHandler stopHandler = getStopHandler();
+                    if (stopHandler != null) {
+                        stopHandler.forceUpdate();
+                    }
+                    break;
+                default:
+                    logger.warn("Unhandled command for channel: {} with comamnd: {}", channelUID.getId(),
+                            command.toString());
```

Comment:
The toString is automaticly called by the logging framework, you should just pass in the command the rest is handled for you